BusBoard.Web/ViewModels/BusInfo.cs

```diff
@@ -1,13 +1,46 @@
-﻿namespace BusBoard.Web.ViewModels
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using BusBoard.Api.Postcode;
+using BusBoard.Api.Tfl;
+using BusBoard.Api.Tfl.Responses;
+
+namespace BusBoard.Web.ViewModels
 {
   public class BusInfo
   {
+    public BusStopInfo BusStop1 { get; }
+    public BusStopInfo BusStop2 { get; }
+
+    public string ErrorMessage { get; } = "";
+
+    private TflApi tflApi = new TflApi();
+    private PostcodeApi postcodeApi = new PostcodeApi();
+
     public BusInfo(string postCode)
     {
-      PostCode = postCode;
-    }
+      try
+      {
+        var postcodeData = postcodeApi.GetPostcodeData(postCode);
+        var stopPoints = tflApi.GetStopPointsAtLocation(postcodeData.latitude, postcodeData.longitude);
+
+        var busStops = stopPoints.Select(stopPoint => new BusStopInfo(stopPoint));
 
-    public string PostCode { get; set; }
+        if (!busStops.Any())
+        {
+          ErrorMessage = "No bus stops near this location.";
+          return;
```

Comment:
Returning early from a constructor is a bit unusual. I'd probably do it as an if/else, but not a massive issue 🙂

Reply:
Sorted in 04fc291 🙂